include/rsg/services/globalService.hpp

```diff
@@ -0,0 +1,31 @@
+#ifndef _RSG_GLOBAL_SERVICE_IMPL_
```

Comment:
When I first began to work on RSG I did not get that the `Engine` was something that really mean something (in the user API). I associated this class with the network purpose.  So in the actual code the Engine is more like the client of RSG  the RSG actor. Thus the Engine you first designed no longer exist and it become the `Global system`. Of course it is just a matter of name (which I understand is very important), so I can rename it, it is not a problem.